src/vs/workbench/browser/parts/views/viewPaneContainer.ts

```diff
@@ -602,13 +613,18 @@ export class ViewPaneContainer extends Component implements IViewPaneContainer {
 
 	protected onDidAddViews(added: IAddedViewDescriptorRef[]): ViewPane[] {
 		const panesToAdd: { pane: ViewPane, size: number, index: number }[] = [];
+
+		const viewContainerRegistry = Registry.as<IViewContainersRegistry>(ViewExtensions.ViewContainersRegistry);
+		const currentLocation = viewContainerRegistry.getViewContainerLocation(this.viewContainer)!;
+
 		for (const { viewDescriptor, collapsed, index, size } of added) {
 			const pane = this.createView(viewDescriptor,
 				{
 					id: viewDescriptor.id,
 					title: viewDescriptor.name,
 					actionRunner: this.getActionRunner(),
-					expanded: !collapsed
+					expanded: !collapsed,
+					location: currentLocation
```

Comment:
why does location need to be passed? `ViewPane` can get the location from `IViewDescriptorService` ?